### ISSUE TITLE
Stable tag 2022.02.02 don't build

### ISSUE THREAD

**Issue by Gaetanbrl** [geosolutions-it/MapStoreExtension#46]
I follow the readme to use 2022.02.02 tag : 

> stable branch will do the same, will follow and submodule revision will be aligned to latest stable branch available

So : 

> node v14.20.1

```
git clone --recursive ...
git submodule update --init
npm install
```

**ERROR with concurrently  (not exists)** as explained [here](https://github.com/geosolutions-it/MapStoreExtension/issues/17#issuecomment-1376960620)

Build with master branch seems worked.

**Comment by catmorales** [comment]
@tdipisa We are still stuck with this issue. Thank you to take it in account.

**Comment by tdipisa** [comment]
@Gaetanbrl thank you for reporting. We will do a quick check as soon as possible to get back to you.

**Comment by MV88** [comment]
@Gaetanbrl 

## TL;DR 
We have released a new version v2022.02.03 only for this repo that solves the build issues present.
I suggest to use this one

## Full explanation

**when you switch** master branch to a release or to another branch you **always** have to re-run
- git submodule update
- npm install

I have tested checking out to v2022.02.02 tag and by running the command above 
Currently I have spotted an error on the backend build see this error
```
$ npm run be:build

> MapStoreExtension@1.0.0 be:build C:\Work\projects\AAA_Geosolutions\MapStoreExtension
> mvn clean install

[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[WARNING] 'parent.relativePath' of POM it.geosolutions.mapstore:mapstore-web:1.0-SNAPSHOT (C:\Work\projects\AAA_Geosolutions\MapStoreExtension\web\pom.xml) points at it.geosolutions.MapStoreExtension:MapStoreExtension-root instead of it.geosolutions.mapstore:mapstore-root, please verify your project structure @ line 4, column 11
[FATAL] Non-resolvable parent POM for it.geosolutions.mapstore:mapstore-web:1.0-SNAPSHOT: Failure to find it.geosolutions.mapstore:mapstore-root:pom:1.0-SNAPSHOT in https://maven.geo-solutions.it was cached in the local repository, resolution will not be reattempted until the update interval of geosolutions has elapsed or updates are forced and 'parent.relativePath' points at wrong local POM @ line 4, column 11
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project it.geosolutions.mapstore:mapstore-web:1.0-SNAPSHOT (C:\Work\projects\AAA_Geosolutions\MapStoreExtension\web\pom.xml) has 1 error[ERROR]     Non-resolvable parent POM for it.geosolutions.mapstore:mapstore-web:1.0-SNAPSHOT: Failure to find it.geosolutions.mapstore:mapstore-root:pom:1.0-SNAPSHOT in https://maven.geo-solutions.it was cached in the local repository, resolution will not be reattempted until the update interval of 
geosolutions has elapsed or updates are forced and 'parent.relativePath' points at wrong local POM @ line 4, column 11 -> [Help 2]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/UnresolvableModelException
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! MapStoreExtension@1.0.0 be:build: `mvn clean install`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the MapStoreExtension@1.0.0 be:build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2023-01-24T10_20_32_980Z-debug.log

```

This can be fixed by changing parent properties of web/pom-xml to 
```xml
<parent>
    <groupId>it.geosolutions.MapStoreExtension</groupId>
    <artifactId>MapStoreExtension-root</artifactId>
    <version>1.0-SNAPSHOT</version>
  </parent>
```

Regarding your errors it may be due to the wrong status you were in and I suggest to check again and **use the new release tag v2022.02.03** we have just released

<hr>

In your comment I also noticed an error regarding mvn not being installed. 

> [backend] 'mvn' n’est pas reconnu en tant que commande interne

this is required and I suggest to install it as otherwise the backend will not be able to start. Details on [maven version here](https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/requirements/#debug-build)

**Comment by Gaetanbrl** [comment]
I had reproduce concurrently error after do : 

1. build mapStoreExtension 2022.02.00 -> npm start  -> stop process
2. With other terminal, build mapStoreExtension master branch -> npm start -> npm stop
3. in new terminal, build mapstoreExtension 2022.02.00 -> concurrently error
4. Come back to mapstoreExtension master -> npm start -> OK he start

I think this is a node error, i dunno how to fix that.

**Comment by MV88** [comment]
> I think this is a node error, i dunno how to fix that.

well it's a lot of back and forth.


- Do everytime you have switched branch run the "git submodule update" and "npm install" as specified above? 
this is key to have correct dependencies installed before starting local env.
- I suggest you to use the latest release available for this repo which is this **v2022.02.03**
- master branch is outdated and there is a pending PR that will update it #44 so I do not suggest to use master right now
- Having so many terminals is not really impacting because if you switch branch on a terminal the local git history of the repository is changed and is reflected everywhere, no matter the terminal you use (just type CTRL + C, to udpate it)
Different is if you have done more clones

**Comment by Gaetanbrl** [comment]
> Do everytime you have switched branch run the "git submodule update" and "npm install" as specified above?

The issue is, for each terminal, i use a different branch (so don't have to update submodule each time), same nodejs version, and never trigger `npm start` at same time (i always kill previous before start other)

**Comment by MV88** [comment]
sorry, but all the terminal are pointing to the same repo/local folder?

**Comment by Gaetanbrl** [comment]
>  all the terminal are pointing to the same repo/local folder

No.

1 version to start = 1 different folder

**Comment by Gaetanbrl** [comment]
Example : 

**Terminal 1**

```
$ cd git/MapStoreExtension-2022.02.00
$ git checkout tags/v2022.02.00
$ git submodule 
-> revision is same as write in tags -> ok no need to update
$ npm install
$ npm start
-> stop service
```

**Terminal 2**

```
$ cd git/MapStoreExtension-master
$ npm install
$ npm start
-> stop service
$ cd ../MapStoreExtension-2022.02.00
$ npm start
-> concurrently error
```

```
$ lsof -i :8081
-> no result -> No service run on 8081
```

**Comment by Gaetanbrl** [comment]
but terminal is just a detail, i could do all in only one, the problem is the same i suppose.
It was like webpack process still exist in the system.

**Comment by MV88** [comment]
you still need to run "git submodule update" to update the revision of the submodule
you may reside on an old revision anyway, even if you do not change branch

a check you can do is to go in the submodule folder (the one aligned with master)
```
cd MapStore2
git log -n 1
```
to compare it with the latest commit here https://github.com/geosolutions-it/MapStore2/commits/master 

moreover do not forget to use this exact command "git submodule update" because I can see you run only "git submodule"


anyway **for now** stick to the latest release of this repo, i.e. v2022.02.03

**Comment by Gaetanbrl** [comment]
Ok @MV88 i had reboot, delete all folder, reinstall and start webpack -> No error.

**Comment by Gaetanbrl** [comment]
> you still need to run "git submodule update" to update the revision of the submodule

yes, thanks, i understand and will use it to force `update submodule` ;) 

> anyway for now stick to the latest release of this repo, i.e. v2022.02.03

noted !

Thanks.

**Comment by Gaetanbrl** [comment]
last question @MV88 , if i upgrade submodule, i will use last submodule.

What if i need to stay align between mapstore2-georchestra/mapstore2 submodule and MapStoreExtension/mapstore2 submodule ?

See this ref issue https://github.com/georchestra/mapstore2-georchestra/issues/591

**Comment by offtherailz** [comment]
Hi,
I added a note here for MapStoreExtension project (because this consideration is valid in general) https://github.com/geosolutions-it/MapStoreExtension/blob/master/README.md#compatibility hope it helps.
